### PR TITLE
fix(segmentation): failing to render SEG because it was on default tool group

### PR DESCRIPTION
**Pull request by wayfarer3130**
### Context
After rehydrating a segmentation, it got applied to the default tool group, which included both volume and non-volume groups, causing an issue on trying to delete or render the segmentation.

### Changes & Results

Updated the UIViewportDialogService to be lower camel case named.  This wasn't complete, but was just annoying.
Added a new default tool group SEGToolGroup that renders the tool group
Use the new default tool group to add the segmentations to so that other windows don't fail to render.

Before the changes, there would be a bunch of logs and failure to show and/or delete the tool groups, and after the segmentation displays correctly and can be deleted.

### Testing
Find a study with a segmentation, with some display sets NOT part of the segmentation, and not volume capable.
Display the study in 4 up, with at least two viewports containing non volume display sets.
Select one of hte non volume display sets
Double click the SEG
Load the SEG
Examine the stack - you shouldn't be seeing exceptions.
Delete the SEG.
It should get deleted.

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Comment by sedghi** [comment]
image

File: extensions/cornerstone-dicom-seg/src/utils/_hydrateSEG.ts [hunk 1]
```diff
@@ -50,13 +63,15 @@ async function _hydrateSEGDisplaySet({
...
 
+
     if (shouldDisplaySeg) {
       viewportGridService.setDisplaySetsForViewport({
         viewportIndex: index,
         displaySetInstanceUIDs: viewport.displaySetInstanceUIDs,
         viewportOptions: {
           viewportType: 'volume',
           toolGroupId,
+          presentationId: null,
```
Comment (sedghi) [inline extensions/cornerstone-dicom-seg/src/utils/_hydrateSEG.ts:74]: it doesn't belong in this PR but I guess it doesn't break anything?

File: extensions/cornerstone-dicom-seg/src/utils/_hydrateSEG.ts [hunk 2]
```diff
@@ -32,6 +32,19 @@ async function _hydrateSEGDisplaySet({
...
+  // Force volume mode on the selected viewport as well as the other viewports
+  // Otherwise, loading a SEG on a non-volume viewport will fail.
+  updatedViewports.forEach(viewport => {
+    viewport.viewportOptions = {
+      viewportType: 'volume',
+      toolGroupId,
+      presentationId: null,
+      initialImageOptions: {
+        preset: 'middle',
+      },
```
Comment (sedghi) [inline extensions/cornerstone-dicom-seg/src/utils/_hydrateSEG.ts:44]: can we refactor this for both of them? (you have one below as well)

File: extensions/cornerstone-dicom-seg/src/viewports/OHIFCornerstoneSEGViewport.tsx [hunk 3]
```diff
@@ -240,6 +243,7 @@ function OHIFCornerstoneSEGViewport(props) {
       toolGroupId,
       extensionManager
     );
+    console.log("* created", toolGroupId);
```
Comment (sedghi) [inline extensions/cornerstone-dicom-seg/src/viewports/OHIFCornerstoneSEGViewport.tsx:246]: say `OHIFSEGViewport .... `

File: extensions/cornerstone-dicom-seg/src/viewports/OHIFCornerstoneSEGViewport.tsx [hunk 4]
```diff
@@ -249,6 +253,7 @@ function OHIFCornerstoneSEGViewport(props) {
         toolGroupId
       );
 
+      console.log("Destroying tool group", toolGroupId);
```
Comment (sedghi) [inline extensions/cornerstone-dicom-seg/src/viewports/OHIFCornerstoneSEGViewport.tsx:256]: ditto

File: extensions/cornerstone-dicom-seg/src/index.tsx [hunk 5]
```diff
@@ -36,9 +38,15 @@ const extension = {
...
-    commandsManager,
-    configuration = {},
-  }) => {},
+    extensionManager,
+  }: Types.Extensions.ExtensionParams): void => {
+    const {
+      toolGroupService
+    } = servicesManager.services;
+    // Create a tool group by default for use with this tool set.
+    createSEGToolGroupAndAddTools(toolGroupService, SEG_TOOLGROUP_DEFAULT, extensionManager);
```
Comment (sedghi) [inline extensions/cornerstone-dicom-seg/src/index.tsx:47]: so we are doing create twice? since we already create inside the SEGViewport

```
toolGroup = createSEGToolGroupAndAddTools(
      toolGroupService,
      toolGroupId,
      extensionManager
    );
    ```

**Review by sedghi — COMMENTED** [review]
small changes requested

File: extensions/cornerstone-dicom-seg/src/utils/_hydrateSEG.ts (hunk 1 above)
Comment (wayfarer3130) [reply extensions/cornerstone-dicom-seg/src/utils/_hydrateSEG.ts:74]: Agreed, it is in the wrong PR, but it doesn't break anything.

File: platform/core/src/services/CustomizationService/CustomizationService.ts [hunk 6]
```diff
@@ -234,19 +234,16 @@ export default class CustomizationService extends PubSubService {
   }
 
   /**
-   * A single reference is either an an array, or a single customization value,
-   * whose id is the id in the object, or the parent id.
-   * This allows for general use to register customizationModule entries.
+   * A single reference is either an string to be loaded from a module,
```
Comment (wayfarer3130) [inline platform/core/src/services/CustomizationService/CustomizationService.ts:237]: The fixes in this file were required in order to test segmentation on the multi-volume view - they were causing infinite recursion due to an incorrect assumption about how the items are defined.

File: platform/viewer/.webpack/webpack.pwa.js [hunk 7]
```diff
@@ -35,7 +35,7 @@ const setHeaders = (res, path) => {
   }
   if (path.indexOf('.pdf') !== -1) {
     res.setHeader('Content-Type', 'application/pdf');
-  } else if (path.indexOf('/frames') !== -1) {
+  } else if (path.indexOf('frames') !== -1) {
```
Comment (wayfarer3130) [inline platform/viewer/.webpack/webpack.pwa.js:38]: Fails on windows, causing e2e server test failures, so I needed this to test locally.

File: platform/viewer/cypress/integration/volume/MPR.spec.js [hunk 8]
```diff
@@ -14,8 +14,9 @@ describe('OHIF MPR', () => {
   });
 
   it('should go MPR for reconstructible displaySets and come back', () => {
-    cy.wait(1000);
+    cy.wait(250);
     cy.get(':nth-child(3) > [data-cy="study-browser-thumbnail"]').dblclick();
+    cy.wait(250);
```
Comment (wayfarer3130) [inline platform/viewer/cypress/integration/volume/MPR.spec.js:19]: Currently intermittent without a bit of wait - would REALLY like to figure out how to read a value in the client to do this instead.

File: platform/viewer/public/config/e2e.js [hunk 9]
```diff
@@ -5,6 +5,7 @@ window.config = {
   showStudyList: true,
   // below flag is for performance reasons, but it might not work for all servers
   omitQuotationForMultipartRequest: true,
+  maxNumberOfWebWorkers: 3,
```
Comment (wayfarer3130) [inline platform/viewer/public/config/e2e.js:8]: OOM on windows due to issues in how the webworkers/WASM is loaded/setup

File: platform/viewer/public/config/multiple.js [hunk 10]
```diff
@@ -9,7 +9,7 @@ window.config = {
     },
   ],
   extensions: [],
-  modes: [],
+  modes: ['@ohif/mode-test'],
```
Comment (wayfarer3130) [inline platform/viewer/public/config/multiple.js:12]: Run in mode-test for the multiple back end so it looks like it does during testing.
